ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/akkastreams/Dispatcher.scala

```diff
@@ -103,17 +135,21 @@ class Dispatcher[Index: Ordering, T] private (
     * Gets all values from start, inclusive, to end, exclusive.
     */
   private def subsource(start: Index, end: Index): Source[(Index, T), NotUsed] =
-    Source
-      .unfoldAsync[Index, (Index, T)](start) { i =>
-        if (i == end) {
-          Future.successful(None)
-        } else {
-          readElement(i).map { t =>
-            val nextIndex = readSuccessor(i, t)
-            Some((nextIndex, (nextIndex, t)))
```

Comment:
we were returning `(nextIndex, t)` which is wrong. The stream should emit the offset of the stored record, not the next one.

Reply:
the wrong behaviour was tested so the test got adapted too. See `DispatcherSpec`.